test/test_cuda.py

```diff
@@ -1733,6 +1733,46 @@ def test_streaming_backwards_device_transfer(self):
         self.assertTrue(a.grad.sum().item() == 4 * size)
         self.assertTrue(b.grad.sum().item() == 4 * size)
 
+    def test_streaming_backward_sync_graph_root(self):
+        # This function tests if bwd ops running on a side stream properly sync with the graph root.
+        fwd_bwd_op_stream = torch.cuda.Stream()
+        bwd_ambient_stream = torch.cuda.Stream()
+        # We need these streams to be different otherwise the test is meaningless.
+        assert fwd_bwd_op_stream != bwd_ambient_stream
+
+        size = int(1e5)
```

Comment:
Why do you need such a big size here? To make sure that the ops are slow enough that you don't have flakyness?

Reply:
That was my first motivation, but it turned out adjusting the size wasn't very effective at exposing the race.  The [`torch.cuda._sleep(int(50 * get_cycles_per_ms()))`](https://github.com/pytorch/pytorch/pull/45787/files#diff-e68a7bc6ba14f212e5e7eb3727394b40R1760) turned out to expose the race much more effectively, but since the runtime of the test depended negligibly on `size`, i didn't change `size`.  I'll reduce it to 1e3, no reason to keep 1e5 now.  For the record I don't think 1e5 matters much for end to end runtime or memory use.

Reply:
Yes I'm sure there is not big reason but we have many many tests. So if all of them use small sizes, it ends up making a visible difference :)

Reply:
changed to 1e3